src/transaction.ts

```diff
@@ -245,25 +245,24 @@ export const sendAtomicTransactionComposer = async function (atcSend: AtomicTran
     } as SendAtomicTransactionComposerResults
     // eslint-disable-next-line @typescript-eslint/no-explicit-any
   } catch (e: any) {
+    Config.logger.info('Received error executing Atomic Transaction Composer, for more information enable the debug flag')
     if (Config.debug && typeof e === 'object') {
       e.traces = []
       Config.logger.debug(
-        'Received error executing Atomic Transaction Composer and debug flag enabled; attempting dry run to get more information',
+        'Received error executing Atomic Transaction Composer and debug flag enabled; attempting simulation to get more information',
       )
-      const dryrun = await performAtomicTransactionComposerDryrun(atc, algod)
-
-      for (const txn of dryrun.txns) {
-        if (txn.appCallRejected()) {
+      const simulate = await performAtomicTransactionComposerSimulate(atc, algod)
+      if (simulate.txnGroups[0].failedAt) {
+        for (const txn of simulate.txnGroups[0].txnResults) {
           e.traces.push({
-            trace: txn.appTrace(),
-            cost: txn.cost,
-            logs: txn.logs,
-            messages: txn.appCallMessages,
+            trace: txn.execTrace,
+            cost: txn.logicSigBudgetConsumed,
```

Comment:
What other fields are there? I suspect we don't want logic sig budget, is there other budget or similar fields we can use?

Reply:
@robdmoore there is no direct cost param in simulate its a bit more granular there is appBudgetConsumed and logicSigBudgetConsumed. We can either do a sum of them and assign as a cost or should we change the structure of the errr itself and instead of cost we will just have appBudgetConsumed and logicSigBudgetConsumed as fields respectively? Cause its no longer dealing with dryrun related outputs and only relying on simulate, might make sense to adjust the error fields to items specific to simulate?

Reply:
I'd use appBudget for this, and maybe add logic app as a separate one if we want it